org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -358,15 +409,9 @@ public void registerWatchers(boolean runInJob) {
 			}
 			List<FileSystemWatcher> fileWatchers = new ArrayList<>();
 			URI formatter = preferenceManager.getPreferences().getFormatterAsURI();
-			if (formatter != null && "file".equals(formatter.getScheme())) {
-				File file = new File(formatter);
-				if (file != null && file.isFile()) {
-					IPath formatterPath = new Path(file.getAbsolutePath());
-					if (!isContainedIn(formatterPath, sources)) {
-						sources.add(formatterPath);
-					}
-				}
-			}
+			addWatcher(formatter, sources);
+			URI settings = preferenceManager.getPreferences().getSettingsAsURI();
+			addWatcher(settings, sources);
 			patterns.addAll(sources.stream().map(ResourceUtils::toGlobPattern).collect(Collectors.toList()));
```

Comment:
If `sources` from above has something like `/home/rgrunber/Scripts/org.eclipse.jdt.core.prefs` in the list then `toGlobPattern` seems to produce `/home/rgrunber/Scripts/org.eclipse.jdt.core.prefs/**` which I'm guessing means the file doesn't get watched.

Reply:
Fixed.